docs/source/Building.md

```diff
@@ -65,20 +65,117 @@ between FlatBuffers and the rest of your project, so issues associated with usin
-# The flatbuffers target carry header search path automatically if CMake > 2.8.11.
-target_link_libraries(own_project_target PRIVATE flatbuffers)
+# The flatbuffers target carries header search path information automatically
+# when using CMake version > 2.8.11.
+target_link_libraries(your_project_target PRIVATE flatbuffers)
 ```
-When build your project the `flatbuffers` library will be compiled and linked 
-to a target as part of your project.
+When building your project, the `flatbuffers` library will then be compiled and linked 
+to the target as part of your project.
+
+#### Automatically Running `flatc`
+
+In addition to the above you will also likely want to add custom CMake targets
+to automatically compile your `*.fbs` files when you build your project.  This
+has the advantage that you will not need to manually run the `flatc` compiler
+(or risk forgetting to do so) when you update your Flatbuffers schema files.
+
+If you've followed the recipe in this section to incorporate Flatbuffers into
+your project's build (recommended) then there will also be a new target called
+`flatc` visible to you; this is the target representing the `flatc` executable
+(compiler) that you will need to run to compile the schemas.  Ideally, whenever
+you build your project, this compiler will be run (and built first if necessary)
+to compile any `*.fbs` files that have changed since the last build.
+
+To set this up you will essentially need to add a series of CMake Custom
+Commands to your build and then make your target depend on the files generated
+by that target.  There are two examples of this in the Flatbuffers package:
+
+1. There is a function provided by the Flatbuffers package called `build_flatbuffers`
+in the file `${FLATBUFFERS_SRC_DIR}/CMake/BuildFlatBuffers.cmake` that does exactly this.
+See the comments at the top of that file for details.  Essentially, you call the function
+with a list of schema files and it will create all of the custom commands for you.
+2. There is also an example of directly adding such custom commands in Flatbuffers'
+own `CMakeLists.txt` file at the root of the source tree (see the function called
+`compile_flatbuffers_schema_to_cpp`).
+
+Since each project has slightly different build requirements, you may want to
+take those two examples and adapt them for your own project's needs, adding
+or removing as necessary.  The first option, if it satifies your project's needs,
+will likely be simpler; the second option will be more flexible.
+
+The following is an example of a simple starting point for the second approach:
+
+1. Create a folder called `flatbuffers` somewhere in your project's source tree.
+2. Place all of your `*.fbs` files in there, along with a `CMakeLists.txt` file.
+3. In the parent folder, add the `flatbuffers` folder via `add_subdirectory`.
+4. In the `flatbuffers/CMakeLists.txt` file put something like this:
+
+```cmake
```

Comment:
I'm not sure what the value is is of putting all this example code in the docs, rather than referring people to where to find it. Putting a lot of script like this in the docs has the disadvantage it will likely get outdated soon.

Reply:
Shall I remove the code then and just keep the couple paragraphs that refer the reader to the examples elsewhere?

Reply:
That would be better I think, yes.